core/cas-server-core-util/src/main/java/org/apereo/cas/util/RegexUtils.java

```diff
@@ -26,31 +23,21 @@ private RegexUtils() {}
      * @return whether this is a valid regex or not
      */
     public static boolean isValidRegex(final String pattern) {
-        try {
-            if (pattern == null) {
-                throw new IllegalArgumentException("Pattern cannot be null");
-            }
-            LOGGER.debug("Pattern {} is a valid regex.", Pattern.compile(pattern).pattern());
-            return true;
-        } catch (final Exception exception) {
-            LOGGER.debug("Pattern {} is not a valid regex.", pattern);
-        }
-        return false;
+        return pattern != null;
```

Comment:
You're totally right. Added tests for the version in master branch, isValidRegex("$$$$") returns true, seems to be a valid regex, but "***" is not.